### PR TITLE
Add support to use different certs, port for each endpoint, run unit tests on ubuntu

### PR DESCRIPTION
**Pull request by chaitanyaenr**
- Prometheus-metrics tool uses the same cert, key and port for all the
  endpoints in the inventory file passed. This will enable it to have
  different certs, port for each endpoint.
- User can define the port, cert, key to be used by a particular
  endpoint in the inventory like host port=prometheus_port 
   cert=/path/to/cert key=/path/to/key
- Add support to run unit tests on ubuntu.

### CODE REVIEW

File: agent/tool-scripts/prometheus-metrics [hunk 1]
```diff
@@ -151,12 +151,8 @@ elif [[ ! -s "$inventory" ]] || [[ ! -f "$inventory" ]]; then
 	echo "Please check if the inventory file exists, is not empty and the inventory variable is not set to a directory"
 	exit 1
 fi
-## set default values for port if not defined
-if [ -z "$port" ]; then
-        port=8443
-fi
 ## parse openshift inventory file
-cat "$inventory" | awk -F " " '{ print $1 }' | sed -n '/^\[masters\]/,/^$/p' | awk '{if (NR!=1) {print}}' | sed '/^$/d' > /run/pbench/inv_hosts
+cat "$inventory" | sed -n '/^\[masters\]/,/^$/p' | awk '{if (NR!=1) {print}}' | sed '/^$/d' > /run/pbench/inv_hosts
```
Comment (ndokos) [inline agent/tool-scripts/prometheus-metrics]: Who guarantees that "/run/pbench" exists? Should this be a temporary file instead?

File: agent/tool-scripts/prometheus-metrics [hunk 2]
```diff
@@ -168,30 +164,40 @@ case "$mode" in
 	;;
 	start)
 	mkdir -p $tool_output_dir/json
-	while read -u 9 master; do
-		# fetch keys from openshift master and set default values for cert and key vars if not defined
+	while read -u 9 file; do
+		# read the input file for host, port, cert and key if any. In case the file doesn't have certs, port and they 
+		# are not defined while registering the tool, the default values are set.
+		# host cert=<cert> key=<key> port=<port>
```
Comment (ndokos) [inline agent/tool-scripts/prometheus-metrics]: This format does not agree with the calculations you do in the next four lines. Which is right?

File: agent/tool-scripts/prometheus-metrics [hunk 3]
```diff
@@ -168,30 +164,40 @@ case "$mode" in
...
 			fi
 			key=/run/pbench/keys/admin.key
 		fi
-		tool_cmd='GOPATH=$pbench_install_dir/.go PATH=$PATH:$GOPATH/bin $script_path/datalog/$tool-datalog $master $interval $tool_log $port $cert $key'
-		metrics_data=$tool_output_dir/$master-stdout.txt
+		tool_cmd='GOPATH=$pbench_install_dir/.go PATH=$PATH:$GOPATH/bin $script_path/datalog/$tool-datalog $host $interval $tool_log $port $cert $key'
+		metrics_data=$tool_output_dir/$host-stdout.txt
 		debug_log "$script_name: running $tool_cmd"
 		eval $tool_cmd >"$metrics_data" 2>"$tool_stderr_file" & echo $! >>$tool_pid_file
 	done 9</run/pbench/inv_hosts
```
Comment (ndokos) [inline agent/tool-scripts/prometheus-metrics]: Why the non-standard descriptor (both here and the 11 in a later while loop?

File: agent/tool-scripts/prometheus-metrics [hunk 4]
```diff
@@ -168,30 +164,40 @@ case "$mode" in
...
-		# fetch keys from openshift master and set default values for cert and key vars if not defined
+	while read -u 9 file; do
+		# read the input file for host, port, cert and key if any. In case the file doesn't have certs, port and they 
+		# are not defined while registering the tool, the default values are set.
+		# host cert=<cert> key=<key> port=<port>
+		host=$(echo $file | awk -F " " '{ print $1 }')
+		port=$(echo $file | grep -w "port" | awk -F " " '{ print $2 }' | awk -F "=" '{print $2}')
+		cert=$(echo $file | grep -w "cert" | awk -F " " '{ print $3 }' | awk -F "=" '{print $2}')
+		key=$(echo $file | grep -w "key" | awk -F " " '{ print $4 }' | awk -F "=" '{print $2}')
+		# fetch keys from openshift master and set default values for port, cert and key vars if not defined
```
Comment (ndokos) [inline agent/tool-scripts/prometheus-metrics]: IIUC, this trashes any settings of port/cert/key through options on the command line: if they are not provided in the inventory file, they are set to the master defaults. So why provide the ability to set these things from the command line at all?

File: agent/tool-scripts/prometheus-metrics (hunk 1 above)
Comment (chaitanyaenr) [inline agent/tool-scripts/prometheus-metrics]: @ndokos /run/pbench directory is created before parsing the file. Also added a check.

File: agent/tool-scripts/prometheus-metrics (hunk 2 above)
Comment (chaitanyaenr) [inline agent/tool-scripts/prometheus-metrics]: @ndokos  calculations are right, I modified the comment to reflect that.

File: agent/tool-scripts/prometheus-metrics (hunk 4 above)
Comment (chaitanyaenr) [inline agent/tool-scripts/prometheus-metrics]: Settings defined from the command line will override the values defined in the file. I modified the script to do that.

File: agent/bench-scripts/unittests [hunk 5]
```diff
@@ -105,10 +105,13 @@ function _verify_output {
...
     # fix up stupidities in python exception output: lib64->lib, line numbers->XXX
-    # fix up Ubuntu stupidities: backtick to single quote
+    # fix up Ubuntu stupidities: 
+    # backtick to single quote
+    # UTF-8 left single quote to ASCII apostrophe
+    # UTF-8 right single quote to ASCII apostrophe
     sed 's/lib64/lib/g
          s/, line [1-9][0-9]*, in/, line XXX, in/g' $_testout |
-         tr '\140' '\047' > $_testout.1
+         tr '\140' '\047' | sed 's/\xe2\x80\x98/\x27/' | sed 's/\xe2\x80\x99/\x27/' > $_testout.1
```
Comment (ndokos) [inline agent/bench-scripts/unittests]: Please split the long pipeline: all you have to do is split it just after each | sign.

**Comment by chaitanyaenr** [comment]
@ndokos PTAL.

**Pull request merged**